docs: accessibility improvements (#4652)
adobe/spectrum-web-components: packages/picker/README.md

import { Picker } from '@spectrum-web-components/picker';
```

### Anatomy

#### Labels

> To render accessibly, an `<sp-picker>` element should be paired with an `<sp-field-label>` element that has a `for` attribute referencing the `id` of the `<sp-picker>` element.

Reviewer: ```suggestion
To render accessibly, an `<sp-picker>` element should be paired with an `<sp-field-label>` element that has the `for` attribute referencing the `id` of the `<sp-picker>` element.
```

</sp-picker>
```

> When you use `<sp-menu-item>` elements without text content, you will need to be sure to use the `value` attribute so that the `<sp-picker>` element can differentiate between the available options. Further, it is important that you apply accessible labeling to the `[slot="icon"]` content as follows:

Reviewer: ```suggestion
When using `<sp-menu-item>` elements without text content, be sure to use the `value` attribute so that the `<sp-picker>` element can differentiate between the available options. Furthermore, it is important to apply accessible labeling to the `[slot="icon"]` content as follows:
```

Removed 2nd person

</sp-picker>
```

##### Advanced icon management

> The `icons` attribute allows you to manage how the selected item will appear. Set `icons` to `only` display only the selected item's icon in the `<sp-picker>` element or `none` to display the selected item text without the icon `<sp-picker>`.

Reviewer: ```suggestion
The `icons` attribute manages how the selected item will appear. Set `icons="only"` to display only the selected item's icon in the `<sp-picker>` element, or `icons="none"` to display the selected item text without the icon `<sp-picker>`.
```

Just a suggestion. Take it or leave it. (Also removed 2nd person)

> When using `icons="only"` and your `<sp-menu-item>` elements still have text content, that content will be applied to `<sp-picker>` element in a non-visible way.

Reviewer: ```suggestion
When using `icons="only"` on `<sp-menu-item>` elements that have text content, that text will be applied to `<sp-picker>` element in a non-visible way.
```

<sp-tabs selected="only" auto label="Icons Attribute Options">
<sp-tab value="only">Only</sp-tab>
<sp-tab-panel value="only">

```html demo
<sp-field-label for="picker-icons-value">Choose an action...</sp-field-label>
<sp-picker

>     label="What would you like to do?"

Author: @jnurthen in the examples they were using the `label` attribute one the picker as placeholder text. If they wanted placeholder text _and_ a label, do we need to rethink picker's API?

>     label="What would you like to do?"

Reviewer 2: When will that ever get shown when there is a default selected value?

>     label="What would you like to do?"

Author: Nope.

So if I look at the [design docs for picker](https://spectrum.adobe.com/page/picker/#Placeholder), it really looks like the `sp-picker`'s `label` attribute was supposed to be a `placeholder`.

I'm going to make an issue out of this one.

    id="picker-icons-value"
    icons="only"
>
    <sp-menu-item value="save">
        <sp-icon-save-floppy slot="icon"></sp-icon-save-floppy>
        Save
    </sp-menu-item>
    <sp-menu-item value="finish">
        <sp-icon-stopwatch slot="icon"></sp-icon-stopwatch>
        Finish
    </sp-menu-item>
    <sp-menu-item value="review">
        <sp-icon-user-activity slot="icon"></sp-icon-user-activity>
        Review
    </sp-menu-item>
</sp-picker>
```

</sp-tab-panel>
<sp-tab value="none">None</sp-tab>
<sp-tab-panel value="none">

```html demo
<sp-field-label for="picker-icons-none">Choose an action...</sp-field-label>
<sp-picker
    label="What would you like to do?"
    value="save"
    id="picker-icons-none"
    icons="none"
>
    <sp-menu-item value="save">
        <sp-icon-save-floppy slot="icon"></sp-icon-save-floppy>
        Save
    </sp-menu-item>
    <sp-menu-item value="finish">
        <sp-icon-stopwatch slot="icon"></sp-icon-stopwatch>
        Finish
    </sp-menu-item>
    <sp-menu-item value="review">
        <sp-icon-user-activity slot="icon"></sp-icon-user-activity>
        Review
    </sp-menu-item>
</sp-picker>
```

</sp-tab-panel>
</sp-tabs>

#### Value

> When the `value` of an `<sp-picker>` matches the `value` attribute or the trimmed `textContent` (or `itemText`) of a descendent `<sp-menu-item>` element, it will make that element as `selected`.

Reviewer: ```suggestion
When the `value` of an `<sp-picker>` matches either the `value` attribute or the trimmed `textContent` (or `itemText`) of a descendent `<sp-menu-item>`, it will mark that element as `selected`.
```

</sp-picker>
```

</sp-tab-panel>
</sp-tabs>

#### Pending

> When in pending state, `<sp-picker>` elements will not respond to click events and will be delivered with a `<sp-progress-circle>` to visually outline that it is pending. It will not toggle open or display its `<sp-menu-item>` descendants until the attribute is removed.

Reviewer: ```suggestion
While in pending state, `<sp-picker>` elements will not respond to click events and will be delivered with an `<sp-progress-circle>` to visually denote that it is pending. It will not toggle open or display its `<sp-menu-item>` descendants until the attribute is removed.
```

> 

Reviewer 2: Should pendingLabel be documented too?

Author: @jnurthen how is this:

```
Use the `pending-label` attribute to customize the pending text for assoistive technology, which is set to `Pending` by default.
```

</sp-picker>
```

</sp-tab-panel>
</sp-tabs>

### Accessibility guidelines

#### Include a visible label

> In rare cases where context is sufficient and a label could be absent, make sure to have the design reviewed and approved by an accessibility expert.

Reviewer: ```suggestion
In rare cases where context is sufficient enough that a label could be absent, make sure to have the design reviewed and approved by an accessibility expert.
```

#### Use help text to show context

A picker’s description in the help text is can communicate what to select or how to select an option. This includes information such as:

-   An overall description of the picker options
-   Hints for what kind of information to choose
-   More context for why a user needs to make a selection

> The help text’s message should not simply restate the same information in the label in order to prompt someone to interact with a picker. Don’t add help text if it isn’t actually relevant or meaningful to a user in order to try to maintain layout continuity with other inputs that require help text.

Reviewer: ```suggestion
The help text’s message should not simply restate the same information in the label in order to prompt someone to interact with a picker. Don’t add help text to maintain layout continuity with other inputs that require help text if it isn’t actually relevant or meaningful to a user.
```
